src/Compilers/CSharp/Portable/Binder/SwitchBinder.cs

```diff
@@ -101,6 +101,18 @@ override protected ImmutableArray<LocalSymbol> BuildLocals()
             return builder.ToImmutableAndFree();
         }
 
+        protected override ImmutableArray<LocalFunctionSymbol> BuildLocalFunctions()
+        {
+            var builder = ArrayBuilder<LocalFunctionSymbol>.GetInstance();
+
+            foreach (var section in _switchSyntax.Sections)
+            {
+                builder.AddRange(BuildLocalFunctions(section.Statements));
```

Comment:
So local functions wouldn't have the same scoping rules as local variables? (defined just above this method)

To be honest, I'm not entirely sure what the scoping rules that fall out from this code are - I just assumed we wanted to do the same thing as local variables. (I think this code was in one of my earlier commits, when I didn't have as good of an idea of how things work as I do now)